### PR TITLE
[aom] Update to 3.11.0

### LINKED ISSUE #41123: [aom] update to 3.10.0
### Library name

aom

### New version number

3.10.0

### Other information that may be useful (release notes, etc...)

Release notes: https://aomedia.googlesource.com/aom/+/refs/tags/v3.10.0/CHANGELOG

### PR DESCRIPTION
**Pull request by jimwang118**
Fixes https://github.com/microsoft/vcpkg/issues/41123

Fix compilation errors under uwp.
`E:\vcpkg\buildtrees\aom\src\a3cfd7c5fc-83bc6a3bee.clean\aom_util/aom_pthread.h(95): error C2664: 'HANDLE CreateThread(LPSECURITY_ATTRIBUTES,SIZE_T,LPTHREAD_START_ROUTINE,LPVOID,DWORD,LPDWORD)': cannot convert argument 3 from 'unsigned int (__cdecl *)(void *)' to 'LPTHREAD_START_ROUTINE'`

- [X] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [X] SHA512s are updated for each updated download.
- [ ] ~~The "supports" clause reflects platforms that may be fixed by this new version.~~
- [ ] ~~Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.~~
- [ ] ~~Any patches that are no longer applied are deleted from the port's directory.~~
- [X] When updating the upstream version, the `"port-version"` is reset (removed from `vcpkg.json`).
- [X] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [X] Only one version is added to each modified port's versions file.

Usage test pass with following triplet:

```
x64-windows
```

### CODE REVIEW

**Review by JonLiu1993 — APPROVED** [review]

File: ports/aom/fix_uwp.patch [hunk 1]
```diff
@@ -0,0 +1,13 @@
+diff --git a/aom_util/aom_pthread.h b/aom_util/aom_pthread.h
+index 89c0c96..eace2c1 100644
+--- a/aom_util/aom_pthread.h
++++ b/aom_util/aom_pthread.h
+@@ -94,7 +94,7 @@ static inline int pthread_create(pthread_t *const thread,
+ #ifdef USE_CREATE_THREAD
+   *thread = CreateThread(NULL,          /* lpThreadAttributes */
+                          0,             /* dwStackSize */
+-                         start, arg, 0, /* dwStackSize */
++                         (LPTHREAD_START_ROUTINE)start, arg, 0, /* dwStackSize */
```
Comment (JavierMatosD) [inline ports/aom/fix_uwp.patch:10]: How are we sure this patch fixes the bug and doesn't just bypass the compilation error? 

The linked issue just includes a link to the change log. Can you provide more context?

Comment (jimwang118) [reply ports/aom/fix_uwp.patch:10]: > How are we sure this patch fixes the bug and doesn't just bypass the compilation error?
> 
> The linked issue just includes a link to the change log. Can you provide more context?

```
E:\vcpkg\buildtrees\aom\src\a3cfd7c5fc-83bc6a3bee.clean\aom_util/aom_pthread.h(95): error C2664: 'HANDLE CreateThread(LPSECURITY_ATTRIBUTES,SIZE_T,LPTHREAD_START_ROUTINE,LPVOID,DWORD,LPDWORD)': cannot convert argument 3 from 'unsigned int (__cdecl *)(void *)' to 'LPTHREAD_START_ROUTINE'
E:\vcpkg\buildtrees\aom\src\a3cfd7c5fc-83bc6a3bee.clean\aom_util/aom_pthread.h(97): note: This conversion requires a reinterpret_cast, a C-style cast or parenthesized function-style cast
C:\Program Files (x86)\Windows Kits\10\include\10.0.22621.0\um\processthreadsapi.h(194): note: see declaration of 'CreateThread'
E:\vcpkg\buildtrees\aom\src\a3cfd7c5fc-83bc6a3bee.clean\aom_util/aom_pthread.h(95): note: while trying to match the argument list '(int, int, unsigned int (__cdecl *)(void *), void *, int, int)'
```
The specific error message is as above. This is a C project code, so C-style forced type conversion is performed.

Comment (BillyONeal) [reply ports/aom/fix_uwp.patch:10]: This cast is not safe. `LPTHREADSTARTROUTINE` is:

```c++
typedef DWORD (__stdcall *LPTHREAD_START_ROUTINE) (  
    [in] LPVOID lpThreadParameter  
); 
```

Note: `__stdcall`. If this were converting from `unsigned int (__stdcall *)(void *)` to `unsigned long (__stdcall *)(void *)` (changing int to long in an environment where int and long are the same size) it *might* be OK, but changing the calling convention with a cast like this is never OK.

**Review by BillyONeal — DISMISSED** [review]

File: ports/aom/portfile.cmake [hunk 2]
```diff
@@ -32,7 +33,6 @@ vcpkg_cmake_configure(
         -DENABLE_TESTDATA=OFF
         -DENABLE_TESTS=OFF
         -DENABLE_TOOLS=OFF
-        -DTHREADS_PREFER_PTHREAD_FLAGS=ON
```
Comment (dg0yt) [inline ports/aom/portfile.cmake:35]: There was a typo. THe proper name is singular `THREADS_PREFER_PTHREAD_FLAG`.

Comment (jimwang118) [reply ports/aom/portfile.cmake:35]: Thanks, it has been modified.

**Review by JonLiu1993 — APPROVED** [review]

File: ports/aom/fix-uwp.patch [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+   (void)attr;
+ #ifdef USE_CREATE_THREAD
+-  *thread = CreateThread(NULL,          /* lpThreadAttributes */
+-                         0,             /* dwStackSize */
+-                         start, arg, 0, /* dwStackSize */
+-                         NULL);         /* lpThreadId */
++  PTP_WORK work = CreateThreadpoolWork((PTP_WORK_CALLBACK)start, arg, NULL);
++  if (work == NULL) return 1;
++  SubmitThreadpoolWork(work);
++  *thread = (pthread_t)work;
```
Comment (BillyONeal) [inline ports/aom/fix-uwp.patch:24]: This is missing `CloseThreadpoolWork`. I think we should just be marking `"supports":"!uwp"`.

Comment (jimwang118) [reply ports/aom/fix-uwp.patch]: Editing completed.

**Comment by jdpurcell** [comment]
And the PR title should be updated to reflect the new version `3.11.0` otherwise the squash merge's commit message will be wrong, no?

@jimwang118 Thank you, by the way!

**Comment by jimwang118** [comment]
> And the PR title should be updated to reflect the new version `3.11.0` otherwise the squash merge's commit message will be wrong, no?
> 
> @jimwang118 Thank you, by the way!

Thanks for your suggestion, I have changed the title.

**Review by JonLiu1993 — APPROVED** [review]

**Pull request merged**